flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/JdbcDialect.java

```diff
@@ -142,4 +143,16 @@
      */
     String getSelectFromStatement(
             String tableName, String[] selectFields, String[] conditionFields);
+
+    /** Create catalog instance. */
+    default AbstractJdbcCatalog createCatalog(
+            String catalogName,
+            String defaultDatabase,
+            String username,
+            String pwd,
+            String baseUrl) {
+        JdbcDialect dialect = JdbcDialectLoader.load(baseUrl);
+        throw new UnsupportedOperationException(
+                String.format("Catalog for '%s' is not supported yet.", dialect));
```

Comment:
This load call is unnecessary, it'll just find itself (`this`) anyway. We're already in the loaded dialect here. We can also just directly use `dialectName`.

```suggestion
                String.format("Catalog for JDBC dialect '%s' is not supported yet.", dialectName()));
```